terraform/full_environment/events_ingestion.tf

```diff
@@ -0,0 +1,188 @@
+# service account for reading data
+resource "google_service_account" "analytics_write" {
+  account_id   = "ga4-write-bq"
+  display_name = "ga4-write-bq"
+  project      = var.gcp_project_id
+}
+
+# custom role for analytics transfer
+resource "google_project_iam_custom_role" "analytics_write_role" {
+  role_id     = "analytics_write_role"
+  title       = "ga4-write-bq-permissions"
+  description = "Write data to vertex schemas in bq"
+
+  permissions = [
+    "bigquery.tables.update",
+    "bigquery.tables.updateData",
+    "bigquery.jobs.create"
+  ]
+}
+
+# binding role to service account
+resource "google_project_iam_binding" "analytics_write" {
+  project = var.gcp_project_id
+  role    = google_project_iam_custom_role.analytics_write_role.id
+
+  members = [
+    google_service_account.analytics_write.member
+  ]
+}
+
+###
+
+# service account for reading ga4 data
+resource "google_service_account" "analytics_read" {
+  account_id   = "ga4-read-bq"
+  display_name = "ga4-read-bq"
+  project      = "GA4-Analytics"
```

Comment:
Instead of hardcoding the project ID, let's add a variable for it after `gcp_project_id` in `variables.tf`:
```terraform
variable "gcp_analytics_project_id" {
  type        = string
  description = "GCP project ID for the project containing BigQuery analytics data"
  default     = "GA4-Analytics"
}
```

And then a data source for it in `events_ingestion.tf`:
```terraform
data "google_project" "analytics_project" {
  project_id = var.gcp_analytics_project_id
}
```

And then refer to it by the data source everywhere, e.g.:
```terraform
resource "google_service_account" "analytics_read" {
  # ...
  project = google_project.analytics_project.project_id
}
```